pkg/kube/qdr/tls_handler.go

```diff
@@ -0,0 +1,206 @@
+package qdr
+
+import (
+	"fmt"
+	"github.com/skupperproject/skupper/api/types"
+	"github.com/skupperproject/skupper/pkg/kube"
+	"github.com/skupperproject/skupper/pkg/qdr"
+	"github.com/skupperproject/skupper/pkg/service"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"strings"
+)
+
+type TlsManagerInterface interface {
+	GetSecret(string) (*corev1.Secret, error)
+	GetConfigMap() (*corev1.ConfigMap, error)
+	NewSecret(types.Credential, *metav1.OwnerReference) (*corev1.Secret, error)
+	DeleteSecret(string) error
+	AddSslProfile(sslProfile qdr.SslProfile) error
+	ExistsSslProfile(string) (bool, error)
+	RemoveSslProfile(string) error
+}
+
+type TlsServiceSupport struct {
+	Address       string
+	Credentials   string
+	CertAuthority string
+}
+
+func EnableTlsSupport(support TlsServiceSupport, tlsManager TlsManagerInterface) error {
```

Comment:
Suggest that this be part of TlsManagerInterface as it appears to be associated to the abstraction

Reply:
Done!